notebooks/source/truncated_distributions.ipynb

```diff
@@ -0,0 +1,1916 @@
+{
```

Comment:
<p>why is <code>x==x</code> necessary?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyro-ppl/numpyro/pull/1272/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

Reply:
Hmmm.. tbh I just copied the pattern from [here](https://github.com/pyro-ppl/numpyro/blob/ed498ab74a6dafdd37ea766e2e8653a51ba129cc/numpyro/distributions/constraints.py#L433). My guess is that it has something to do with NaN values, but not 100% sure.

Reply:
<p>That's right. NaN is an invalid value.</p>